flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/handlers/JarHandlerParameterTest.java

```diff
@@ -69,6 +71,11 @@ enum ProgramArgsParType {
     static final String[] PROG_ARGS = new String[] {"--host", "localhost", "--port", "1234"};
     static final int PARALLELISM = 4;
 
+    static final Map<String, String> FLINK_CONFIGURATION =
+            ImmutableMap.of(
+                    CoreOptions.DEFAULT_PARALLELISM.key(), "2",
```

Comment:
needs a test for what happens if this isn't set in the config, since we have special logic for this case.

Reply:
Sure, I think unit tests should cover all requested values, including values ​​like: 

- DEFAULT_PARALLELISM
- RESTORE_MODE
- SAVEPOINT_IGNORE_UNCLAIMED_STATE
- SAVEPOINT_PATH

what do you think?

Reply:
I only insist on testing the parallelism setting; the rest is up to you.

Reply:
I found that this part cannot directly test only the parallelism part, because when we run JarRunHandler in the code, the above value will still appear in flinkconfig, so I overwrite the value of this part during unit testing, which is why I re- add the `JarHandlerParameterTest.testJarRequestBodyConfigEqToHandlerParameters` method

Reply:
> when we run JarRunHandler in the code, the above value will still appear in flinkconfig

I don't get it. We are in full control in these tests as to whether the configuration is passed via the request body and what "flink configuration" is passed to the handler at construction time.
Why would it _impossible_ to test?

Reply:
because both **JarPlanHandler** and **JarRunHandler** are tested, but in fact, JarPlanHandler has nothing to do with Savepoint settings, so I extracted a method here. If I need to improve this part of the test to make it simple, what should I do, this is where I'm confused 🤔

Reply:
Now I do this with a new validation with JobGraph method